### PR TITLE
ansifilter: add @ 1.15

### PR DESCRIPTION
**Pull request by Adjective-Object**
###### Things done:
- [x] Tested using sandboxing (`nix-build --option build-use-chroot true` or [nix.useChroot](http://nixos.org/nixos/manual/options.html#opt-nix.useChroot) on NixOS)
- Built on platform(s)
  - [x] NixOS
  - [ ] OS X
  - [ ] Linux
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).
###### More

added package [ansifilter](http://www.andre-simon.de/doku/ansifilter/en/ansifilter.php) @ 1.15

### CODE REVIEW

File: pkgs/tools/text/ansifilter/default.nix [hunk 1]
```diff
@@ -0,0 +1,22 @@
+{ stdenv, pkgs }:
+let version = "1.15";
+    pkgsha = "65dc20cc1a03d4feba990f830186404c90462d599e5f4b37610d4d822d67aec4";
+in with pkgs; stdenv.mkDerivation {
+  name = "ansifilter";
```
Comment (joachifm) [inline pkgs/tools/text/ansifilter/default.nix]: Please make version a part of the name, as in `name = ansifilter-${version}`.

File: pkgs/tools/text/ansifilter/default.nix [hunk 2]
```diff
@@ -0,0 +1,22 @@
+{ stdenv, pkgs }:
```
Comment (joachifm) [inline pkgs/tools/text/ansifilter/default.nix]: Please pass in specific members of `pkgs`, passing in the whole package set is bad practice.

File: pkgs/tools/text/ansifilter/default.nix [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+let version = "1.15";
+    pkgsha = "65dc20cc1a03d4feba990f830186404c90462d599e5f4b37610d4d822d67aec4";
+in with pkgs; stdenv.mkDerivation {
+  name = "ansifilter";
+  buildInputs = [
+    stdenv gcc gnumake
```
Comment (joachifm) [inline pkgs/tools/text/ansifilter/default.nix]: `stdenv`, `gcc`, and `gnumake` are redundant build inputs (the point of stdenv is to provide an environment where essential build commands are available).

File: pkgs/tools/text/ansifilter/default.nix [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    pkgconfig boost lua
+  ];
+  src = fetchurl {
+    url = "http://www.andre-simon.de/zip/ansifilter-${version}.tar.bz2";
+    sha256 = pkgsha;
+  };
+  meta = {
```
Comment (joachifm) [inline pkgs/tools/text/ansifilter/default.nix]: The `meta` attribute needs to be fleshed out; at least provide license and a proper description.

File: pkgs/tools/text/ansifilter/default.nix [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    description = "tool";
+  };
+
+  configurePhase = ''makeFlags="PREFIX=$out conf_dir=$out/etc/ansifilter/"'';
+
+  maintainers = [ lib.maintainers.adjective ];
```
Comment (joachifm) [inline pkgs/tools/text/ansifilter/default.nix]: This should be a member of the `meta` attribute

File: pkgs/tools/text/ansifilter/default.nix [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+  ];
+  src = fetchurl {
+    url = "http://www.andre-simon.de/zip/ansifilter-${version}.tar.bz2";
+    sha256 = pkgsha;
+  };
+  meta = {
+    description = "tool";
+  };
+
+  configurePhase = ''makeFlags="PREFIX=$out conf_dir=$out/etc/ansifilter/"'';
```
Comment (joachifm) [inline pkgs/tools/text/ansifilter/default.nix]: You can provide additional make flags by specifying a `makeFlags` attribute, as in `makeFlags = "conf_dir=$(out)/etc/ansifilter"`.

**Comment by joachifm** [comment]
Please change the commit subject to `ansifilter: init at 1.5`.

File: lib/maintainers.nix [hunk 7]
```diff
@@ -12,6 +12,7 @@
   abbradar = "Nikolay Amiantov <ab@fmap.me>";
   aboseley = "Adam Boseley <adam.boseley@gmail.com>";
   adev = "Adrien Devresse <adev@adev.name>";
+  adjective = "Maxwell Huang-Hobbs <mhuan13@gmail.com>";
```
Comment (joachifm) [inline lib/maintainers.nix]: It'd be nice if you'd re-use your GitHub handle here

**Comment by Adjective-Object** [comment]
Sorry about the description! That's embarrasing.

I implemented the changes you suggested, apart from the makeFlags change. I encountered a weird error where the package build wouldn't actually produce anything if I specified the make flags with makeFlags instead of by setting the environment variables as I am now.

This method of passing makeFlags is modeled after the [highlight](https://github.com/NixOS/nixpkgs/blob/788800e437c4a0a25d95e217540bded68804b25e/pkgs/tools/text/highlight/default.nix) package, from the same people who made ansifilter.

**Comment by joachifm** [comment]
Did you try the `$(out)` syntax for makeFlags? They are being passed verbatim to make, so expansion occurs within make, not the shell (thus, `$out` won't work).

File: lib/maintainers.nix [hunk 8]
```diff
@@ -12,6 +12,7 @@
   abbradar = "Nikolay Amiantov <ab@fmap.me>";
   aboseley = "Adam Boseley <adam.boseley@gmail.com>";
   adev = "Adrien Devresse <adev@adev.name>";
+  adjective-object = "Maxwell Huang-Hobbs <mhuan13@gmail.com>";
```
Comment (joachifm) [inline lib/maintainers.nix]: Please make it the exact same as your github handle; this is to make it easier to ping you on github if necessary

**Comment by Adjective-Object** [comment]
Updated maintainer name and used `$(out)` syntax, sorry for the trouble

File: pkgs/tools/text/ansifilter/default.nix [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+  name = "ansifilter-${version}";
+  buildInputs = [
+    pkgconfig boost lua
+  ];
+  src = fetchurl {
+    url = "http://www.andre-simon.de/zip/ansifilter-${version}.tar.bz2";
+    sha256 = pkgsha;
+  };
+  meta = {
+    description = ''
```
Comment (joachifm) [inline pkgs/tools/text/ansifilter/default.nix]: Minor nitpick: please remove the newline at the start here; if you want to provide a longer form description, use `longDescription`.

**Comment by joachifm** [comment]
No trouble. Apart from the final nitpick, I think this is ready to merge.

**Comment by joachifm** [comment]
Merged, thanks

**Pull request merged**